### ISSUE TITLE
bug in Modal component with version 3.3.5

### ISSUE THREAD

**Issue by happy2deepak** [twbs/bootstrap#18100]
I'm very new to Github issue reporting. So, do let me know, if anything needs to be explained better.
Issue:
Suppose there are two modals, modal-1 and modal-2 having link to each other on the modal itself.

The navigation is handled using below code. data-dismiss is added to close the current modal and then open the next modal.

<button class="btn btn-primary" data-target="modal-2" data-toggle="modal" data-dismiss="modal">Modal 2</button>

When you browse to modal-2 from modal-1, the modal-open class gets removed from the body tag and isn't added back. Which causes overflow:hidden and user is unable to scroll the page.

For better understanding, please look these examples with different versions. Click on Modal 1, and traverse to modal-2 from modal-1. And try to scroll everytime a modal appears.

Ver 3.3.5
http://codepen.io/happy2deepak/pen/ojdKOo?editors=100

ver. 3.2.0
http://codepen.io/happy2deepak/pen/GpdawK?editors=100

**Comment by twbs-lmvtfy** [comment]
Hi @happy2deepak!

You appear to have posted a live example (http://codepen.io/happy2deepak/pen/GpdawK.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 9, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 10, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 16, column 9: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 32, column 9: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by twbs-lmvtfy** [comment]
Hi @happy2deepak!

You appear to have posted a live example (http://codepen.io/happy2deepak/pen/ojdKOo.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 9, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 10, column 3: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 16, column 9: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.
* line 32, column 9: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by happy2deepak** [comment]
I don't how to add a revised version. But I have fixed the issue and adding the links again. The matter is still same.

Ver 3.3.5
http://codepen.io/happy2deepak/pen/ojdKOo?editors=100

ver. 3.2.0
http://codepen.io/happy2deepak/pen/GpdawK?editors=100

**Comment by twbs-lmvtfy** [comment]
Hi @happy2deepak!

You appear to have posted a live example (http://codepen.io/happy2deepak/pen/ojdKOo.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 16, column 9: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by happy2deepak** [comment]
Bootstrap usage erros have appeared. Its still showing some fix issue. So pasting the example links again.
Ver 3.3.5
http://codepen.io/happy2deepak/pen/ojdKOo?editors=100

ver. 3.2.0
http://codepen.io/happy2deepak/pen/GpdawK?editors=100

**Comment by twbs-lmvtfy** [comment]
Hi @happy2deepak!

You appear to have posted a live example (http://codepen.io/happy2deepak/pen/ojdKOo.html), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 23, column 9 thru column 111: Attribute ```` href ```` not allowed on element ```` button ```` at this point.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by happy2deepak** [comment]
fixed the button="type" button error and pasting the example URL again.

Ver 3.3.5
http://codepen.io/happy2deepak/pen/ojdKOo?editors=100

ver. 3.2.0
http://codepen.io/happy2deepak/pen/GpdawK?editors=100

**Comment by twbs-lmvtfy** [comment]
Hi @happy2deepak!

You appear to have posted a live example (http://codepen.io/happy2deepak/pen/ojdKOo.html), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 16, column 9: [W007](https://github.com/twbs/bootlint/wiki/W007): Found one or more `<button>`s missing a `type` attribute.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
Duplicate of #17956.
